doc/expressionLanguage.md

```diff
@@ -175,3 +176,35 @@ The following table shows expressions and their expected outcomes for a measure
 | '@value * 5.2'              | 31.2                  |
 | '"Pruebas " + "De Strings"' | 'Pruebas De Strings'  |
 | '@name value is @value'     | 'DevId629 value is 6' |
+
+## <a name="ngsiv2"/> NGSIv2 support
+
+As it is explained in previous sections, expressions can have two return types: String or Number, being the former one the default. Whenever an expression is executed without error, its result will be cast to the configured type. 
+
+On one hand, in NGSIv1 since all attributes' values are of type String, in the expression parser the expression type is set always to String and the transformation of the information coming from the SouthBound is done using replace instruction. Therefore, values sent to the CB will always be Strings. This can be seen in (#execution) example.
+
+On the other hand, NGSIv2 fully supports all the types described in the JSON
+specification (string, number, boolean, object, array and null). Therefore, the result of an expression must be cast to the appropriate type (the type used to define the attribute) in order to avoid inconsistencies between the type field for an attribute and the type of the value that is being sent.
+
+Currently, the expression parser does not support JSON Arrays and JSON document. A new issue has been created to address this aspect https://github.com/telefonicaid/iotagent-node-lib/issues/568. For the rest of types the workflow will be the following:
+
+1. Variables will be cast to Number or String depending on the expression type.
+2. The expression will be applied
+3. The output type will be cast again to the original attribute type.
```

Comment:
As sub-bullets of 3, I would suggest to add the following text in order to be completly splicit about how the cast is done (please, check if it correct... I could have understood something incorrectly):

* If attribute type is "Integer" then the value is casted to integer (JSON number)
* If attribute type is "Float" then the value is casted to float (JSON number)
* If attribute type is "Boolean" then the value is cast to boolean (JSON boolean)
* If attribute type is "Null" then the value is cast to null (JSON null)

Reply:
Fixed in 4026d75714dd6ef51230bd01c8482c0a89e36923